### ISSUE TITLE
Canvas: Release Beta

### ISSUE THREAD

**Issue by nmarrs** [grafana/grafana#40032]
Release canvas panel beta. This means that canvas panel should be mostly devoid of bugs and add value for common edge / IoT use cases that we provide a few basic simulations for

Items marked with [CONTINGENT ON FRAME SUPPORT] are not necessary for initial beta release and may or may not be implemented depending on whether or not we move forward with nested frames in future (pending community feedback)

### ⭐️  Key priority projects
- [x] #47736
- [x] #47740
- [x] #47742

### 👀 Nice to haves
- [x] #55121
- [x] #55328
- [x] #55329
- [x] #52492
- [x] #55330
- [ ] #55331
- [ ] Ability to draw lines between elements (https://raintank-corp.slack.com/archives/C01FJF3LESW/p1650675741368829)
- [ ] #52823

### 🧼 Polishing tasks 
- [x] #52822
- [x] #56138

### 🐞 Bugs to squash 

- [x] #47669
- [x] #48386
- [x] #48388
- [x] #48389
- [x] Glitch when moving multiselection with element having center constraint (see #51270)
- [x] #52116
- [x] #52946
- [x] #52948
- [x] #52951
- [x] #52954
- [x] #55135
- [x] #55689

### 🔮 Future ideas (past beta)
- [CONTINGENT ON FRAME SUPPORT] Nested frame / group support (leveraging Figma's paradigm)
  - Create group concept (currently only have frame implemented)
  - Way to draw / create frame
  - Moving elements between frames (layers) (based on mouse position)
  - Double clicking lets you select inner element inside frame
  - Ability to toggle `Clip content` (overflow) and address UX of hidden elements (dragging etc.)
    - A few ideas if any issues: When one of frame's element is selected turn on overflow and set opacity of "hidden" elements to ~20%. In addition to removing the overflow hidden css, you could also add a transparent box-shadow around the frame div, with a very large width, essentially should "gray out" everything outside of the frame.
  - Context menu option to select nested frames (behind current selected frame)
- Adding useful tools / libraries to augment functionality (such as library that helps with precise placement of elements inside a grid etc.)
- Preview of elements (maybe preloaded with data similar to suggestions feature for adding new panel?)
- Support SVG + custom animations based on data
- onClick support (figuring out sustainable approach to write interaction via invoking custom API calls) (ideally should support both REST + GraphQL as part of this) -> initially could have this onClick support for icon / button elements 
- Data driven layer visibility (whether or not element / group is visible or not)
  - Lowest hanging fruit here being able to hide individual elements based on data 
- Additional element types that can be added (buttons, box, text, icon) (match other example) (epic in itself)
  - Gauge (Grafana)
  - Stat panel (count) (draw)
  - Level indicator (bar, resize reshape like temp, tank etc.)
  - Toggles
    - Checkbox (draw)
    - Sliders (draw)
    - On / off
  - drop down select (draw)
  - SVG (be able to animate speed etc.)
  - Range slider
  - 3 way toggle (on / idle / off)
- Select group options (similar to selected elements having option panel, we should probably have one for group as well)
- Address wind turbine / spinny things animation stutter issue at low rpms with live pipeline (low priority)
- [MINOR] When trying to screencast on macOS, we keep applying `shift` so user is unable to unselect / forces into multiselect mode due to `toggleContinueSelect: 'shift'`
  - https://user-images.githubusercontent.com/22381771/178844159-647724fe-4216-4013-8bba-bdeb0239638e.mov
- #48229
- #48225

### 🏢 Previous meetings
 - [Meeting notes (all)](https://docs.google.com/document/d/18JHp_M5bWIkuyZuUIrAsk7xukHORVwCJCwVcPDZbgD0/edit#)

### 📚 Past issues 
- [x] #40033
- [x] #40348
- [x] #40417
- [x] #40648
- [x] #40732
- [x] #42095
- [x] #41060
- [x] #42401
- [x] #41158
- [x] #42693
- [x] #43121
- [x] #43237 
- [x] #43253

### 🐞 9.2.0 Release testing/feedback

- [ ] #56512

**Comment by grafanabot** [comment]
This issue was removed from the 9.2.0 milestone because 9.2.0 is currently being released.

**Comment by nmarrs** [comment]
Closing this issue out as canvas was released as beta in v9.2 🎉
